Content.Server/_Starlight/Shadekin/ShadekinSystem.Brighteye.cs

```diff
@@ -0,0 +1,244 @@
+                if (_tag.HasTag(spawnmap.Value, _theDarkTag))
+                    spawns.Add(spawnUid);
+
+        // If no valid spawnpoint... we just... DIE!
+        if (spawns.Count <= 0)
+            return;
+
+        _random.Shuffle(spawns);
+
+        // First, Drop Everything we have.
+        if (TryComp<InventoryComponent>(uid, out var inventoryComponent) && _inventorySystem.TryGetSlots(uid, out var slots))
+            foreach (var slot in slots)
+                _inventorySystem.TryUnequip(uid, slot.Name, true, true, false, inventoryComponent);
+
+        // Spawn the Shadow.
+        SpawnAtPosition(_shadekinShadow, Transform(uid).Coordinates);
+
+        // Teleport to "The Dark"
+        foreach (var spawnUid in spawns)
+        {
+            _transform.SetCoordinates(uid, Transform(spawnUid).Coordinates);
+            break;
+        }
+
+        var effect = SpawnAtPosition(_shadekinPhaseInEffect2, Transform(uid).Coordinates);
+        Transform(effect).LocalRotation = Transform(uid).LocalRotation;
+
+        RaiseLocalEvent(uid, new RejuvenateEvent());
+        _sleeping.TrySleeping(uid);
+
+        component.Energy = 0;
+        component.Rejuvenating = true;
+        _alerts.ShowAlert(uid, component.RejuvenationAlert);
+        Dirty(uid, component);
+    }
+
+    /// <summary>
+    /// Change the humanoid eye to be bright and glow!
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="humanoid"></param>
+    public void SetBrighteyes(EntityUid uid, HumanoidAppearanceComponent humanoid)
+    {
+        var hsv = Color.ToHsv(humanoid.EyeColor);
+        hsv.Z = 1.0f;
+        humanoid.EyeColor = Color.FromHsv(hsv);
+
+        humanoid.EyeGlowing = true;
+        Dirty(uid, humanoid);
+    }
+
+    /// <summary>
+    /// Change the humanoid eye to be validated by HumanoidEyeColor.Shadekin (Blackeyes)
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="humanoid"></param>
+    public void SetBlackeyes(EntityUid uid, HumanoidAppearanceComponent humanoid)
+    {
+        humanoid.EyeColor = EyeColor.MakeShadekinValid(humanoid.EyeColor);
+        humanoid.EyeGlowing = false;
+
+        Dirty(uid, humanoid);
+    }
+
+    /// <summary>
+    /// When triggered, will check if we have enough energy and if yes drain the energy and return the value.
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="component"></param>
+    /// <param name="cost">cost of energy (if null then no cost needed)</param>
+    /// <returns></returns>
+    public bool OnAttemptEnergyUse(EntityUid uid, BrighteyeComponent component, int? cost = null)
+    {
+        var ev = new OnAttemptEnergyUseEvent(uid);
+        RaiseLocalEvent(uid, ev);
+
+        if (ev.Cancelled)
+            return false;
+
+        if (cost is null)
+            return true;
+
+        if (component.Energy >= cost)
+        {
+            component.Energy -= (int)cost;
```

Comment:
nit: its already int, no need to cast it

```suggestion
            component.Energy -= cost;
```